src/vm/docs/mod.rs

```diff
@@ -581,10 +581,13 @@ const AT_BLOCK: SpecialAPI = SpecialAPI {
     description: "The `at-block` function evaluates the expression `expr` _as if_ it were evaluated at the end of the
 block indicated by the _block-hash_ argument. The `expr` closure must be read-only.
 
+Note: The block identifying hash must be a hash returned by the `id-header-hash` block information
+property. This hash uniquely identifies Stacks blocks, whereas `header-hash` is not necessarily unique.
+
 The function returns the result of evaluating `expr`.
 ",
     example: "(at-block 0x0000000000000000000000000000000000000000000000000000000000000000 (var-get data))
-(at-block (get-block-info? header-hash (- block-height u10)) (var-get data))"
+(at-block (get-block-info? id-header-hash (- block-height u10)) (var-get data))"
```

Comment:
Not a strong opinion, but I think that passing a block-height to `(at-block ...)` would make this feature more accessible to smart contract developers (+ more terse).